presentation/src/main/java/com/dcs/presentation/trend/TrendRoute.kt

```diff
@@ -0,0 +1,213 @@
+                title = stringResource(id = R.string.movie_trend_title_trend),
+                tabs = MovieTrendType.entries.map { it.toUiString() },
+                movies = viewModel.moviesByTrending
+            ) { tabIndex ->
+                viewModel.updateMovieTrendType(MovieTrendType.entries[tabIndex])
+            }
+
+            TrendMoviePanel(
+                title = stringResource(id = R.string.movie_trend_title_popular),
+                tabs = MoviePopularType.entries.map { it.toUiString() },
+                movies = viewModel.moviesByPopular
+            ) { tabIndex ->
+                viewModel.updateMoviePopularType(MoviePopularType.entries[tabIndex])
+            }
+
+            TrendMoviePanel(
+                title = stringResource(id = R.string.movie_trend_title_upcoming),
+                movies = viewModel.moviesByUpcoming
+            )
+        }
+    }
+}
+
+@Composable
+fun TrendMoviePanel(
+    modifier: Modifier = Modifier,
+    title: String,
+    tabs: List<String> = listOf(),
+    movies: StateFlow<UiState<StateFlow<PagingData<MovieItemUiState>>>>,
+    onTabClick: (Int) -> Unit = {}
+) {
+    var selectedTabIndex by remember { mutableIntStateOf(0) }
+
+    Column(
+        modifier = modifier
+            .fillMaxWidth()
+    ) {
+        Text(
+            modifier = Modifier.padding(all = 12.dp),
+            text = title,
+            fontWeight = FontWeight.Bold,
+            fontSize = 18.sp,
+            color = Color.Black
+        )
+
+        if (tabs.isNotEmpty()) {
+            CustomScrollableTabRow(
+                tabs = tabs,
+                selectedTabIndex = selectedTabIndex,
+            ) { tabIndex ->
+                selectedTabIndex = tabIndex
+                onTabClick.invoke(tabIndex)
+            }
+        }
+
+        TrendMovieUiStateScreen(movies)
+
+        HorizontalDivider(
+            Modifier.padding(all = 4.dp),
+            color = Color.LightGray
+        )
+    }
+}
+
+@Composable
+fun CustomScrollableTabRow(
+    tabs: List<String>,
+    selectedTabIndex: Int,
+    onTabClick: (Int) -> Unit
+) {
+    ScrollableTabRow(
+        selectedTabIndex = selectedTabIndex,
+        contentColor = Color.Black,
+        edgePadding = 0.dp,
+        divider = {},
+        modifier = Modifier
+            .wrapContentSize(),
+    ) {
+        tabs.forEachIndexed { tabIndex, tab ->
+            val selected = selectedTabIndex == tabIndex
+            val textColor = if (selected) Color.DarkGray else Color.LightGray
+            Tab(
+                selected = selected,
+                onClick = { onTabClick(tabIndex) },
+                text = {
+                    Text(
+                        text = tab,
+                        color = textColor
+                    )
+                }
+            )
+        }
+    }
+}
+
+@Composable
+fun TrendMovieUiStateScreen(
+    movies: StateFlow<UiState<StateFlow<PagingData<MovieItemUiState>>>>,
+) {
```

Comment:
```suggestion
fun TrendMovieUiStateScreen(
    movies: StateFlow<UiState<StateFlow<PagingData<MovieItemUiState>>>>,
    modifier: Modifier = Modifier
) {
```

외부에서 해당 컴퍼넌트를 조정할 수 있도록 modifer를 추가하는게 좋을 것 같아요

Reply:
적용 완료